src/rpc/handlers/LedgerData.cpp

```diff
@@ -73,6 +73,10 @@ doLedgerData(Context const& context)
         return *status;
 
     auto lgrInfo = std::get<ripple::LedgerInfo>(v);
+
+    if (!context.backend.fetchLedgerObject(cursor, lgrInfo.seq))
```

Comment:
I think `cursor` should be marker. Also, let's only do this check if `outOfOrder` is false. And lastly, we should add some additional message to the response warning about reusing markers from different servers